pkg/archive/archive_unix.go

```diff
@@ -57,6 +58,10 @@ func minor(device uint64) uint64 {
 // handleTarTypeBlockCharFifo is an OS-specific helper function used by
 // createTarFile to handle the following types of header: Block; Char; Fifo
 func handleTarTypeBlockCharFifo(hdr *tar.Header, path string) error {
+	if shared.RunningInUserNS() {
```

Comment:
we shouldn't have to import a lib for this functionality though :) this shouldnt be hard to test by checking proc or something